api/include/opentelemetry/trace/span_context.h

```diff
@@ -41,7 +41,7 @@ class SpanContext final
               SpanId span_id,
               TraceFlags trace_flags,
               bool is_remote,
-              nostd::shared_ptr<TraceState> trace_state = TraceState::GetDefault()) noexcept
+              const nostd::shared_ptr<TraceState> &trace_state = TraceState::GetDefault()) noexcept
```

Comment:
Class SpanContext is used almost everywhere.

Not sure if this change breaks the ABI or not, this needs more analysis.

Reply:
I think there would be a way to compromise on the direct requirement from cppcheck in order to ensure ABI compatibility. I reverted this change (pass by const ref -> pass by value) and move the copy of that shared_ptr into the member variable. 

This should be comparable since it doesn't waste making an extra copy and incrementing/decrementing the use count. 
Verified that cppcheck agrees with this change and no longer reports it as problematic.